jujugui/static/gui/src/app/models/handlers.js

```diff
@@ -301,12 +334,18 @@ YUI.add('juju-delta-handlers', function(Y) {
       // Process the stream.
       db.services.process_delta(action, data);
       if (action !== 'remove') {
-        db.services.getById(change.name).updateConfig(change.config || {});
+        const app = db.services.getById(change.name);
+        // TODO frankban: what is this for?
+        app.updateConfig(change.config || {});
         // Execute the registered application hooks.
         var hooks = applicationChangedHooks[change.name] || [];
         hooks.forEach(function(hook) {
           hook();
         });
+        // Handle GUI additional services.
+        app.get('units').each(unit => {
```

Comment:
Why is this being called both in application and unit deltas?

Reply:
Because we need to react to changes in both entities. An application can be exposed, a unit can be created or removed.